src/plugins/intel_gpu/src/graph/graph_optimizer/remove_redundant_reorders.cpp

```diff
@@ -421,6 +421,132 @@ void remove_redundant_reorders::run(program& p) {
+                auto can_omit_padding = ((output_layout.format == format::b_fs_yx_fsv16 || output_layout.format == format::b_fs_yx_fsv32) &&
+                                        (prev.get_output_layout().format == format::bfyx || prev.get_output_layout().format == format::b_fs_yx_fsv4)) ||
+                                        (output_layout.format == format::b_fs_zyx_fsv16 && prev.get_output_layout().format == format::bfzyx);
+
+                if (output_padded && !can_omit_padding) {
+                    if (prev.get_users().size() != 1)
+                        continue;
+
+                    if (prev.is_type<input_layout>())
+                        continue;
+
+                    prev.merge_output_padding(output_layout.data_padding);
+                }
+
+                node.can_be_optimized(true);
+                LOG_NODE_REMOVAL(node.id());
+                p.extract_and_remove(node);
+
+                if (next_ptr->is_type<fully_connected>())
+                    next_ptr->recalc_output_layout(true);
+            }
+
+            // Fuse reorder into previous node
+            if (prev.is_type<eltwise>() || prev.is_type<fully_connected>()) {
+                bool can_fuse_reorder = false;
+                if (prev.is_type<fully_connected>() && lo.get_optimization_attributes().use_onednn_impls
+                    && output_layout.format == prev.get_preferred_output_fmt()
+                    && output_layout.data_padding == prev.get_output_layout().data_padding) {
+                    auto check_data_types_for_fully_connected = [](data_types in_dt, data_types wei_dt, data_types out_dt) {
+                        if ((in_dt == data_types::f16 && wei_dt == data_types::f16) &&
+                            (out_dt == data_types::f16 || out_dt == data_types::f32 || out_dt == data_types::i8))
+                            return true;
+                        if (in_dt == data_types::f32 && wei_dt == data_types::f32)
+                            return true;
+                        if ((in_dt == data_types::i8 || in_dt == data_types::u8) && (wei_dt == data_types::i8) &&
+                            (out_dt == data_types::i8 || out_dt == data_types::u8 ||
+                            out_dt == data_types::i32 || out_dt == data_types::f16 || out_dt == data_types::f32))
+                            return true;
+                        return false;
+                    };
+
+                    auto src_dtype = prev.as<fully_connected>().input().get_output_layout().data_type;
+                    auto weight_dtype = prev.as<fully_connected>().weights().get_output_layout().data_type;
+                    // Check data type combination for oneDNN inner product
+                    if (check_data_types_for_fully_connected(src_dtype, weight_dtype, output_layout.data_type))
+                        can_fuse_reorder = true;
+                }
+
+                if (prev.is_type<eltwise>() && (fmt_prev == fmt_next))
```

Comment:
why do you compare against fmt_next? I think this should be `output_layout.format`